src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -260,6 +260,56 @@ public TypeSymbolWithAnnotations SetIsAnnotated(CSharpCompilation compilation)
         public TypeSymbolWithAnnotations AsNullableReferenceType() => _extensions.AsNullableReferenceType(this);
         public TypeSymbolWithAnnotations AsNotNullableReferenceType() => _extensions.AsNotNullableReferenceType(this);
 
+        public (bool, TypeSymbolWithAnnotations) TryMergeNullability(TypeSymbolWithAnnotations other, VarianceKind variance)
+        {
+            (bool isNullableSuccess, bool? isNullable) = TryMergeIsNullable(IsNullable, other.IsNullable, variance);
+            if (isNullableSuccess)
+            {
+                (bool typeSuccess, TypeSymbol type) = TypeSymbol.TryMergeNullability(other.TypeSymbol, variance);
+                if (typeSuccess)
+                {
+                    return (true, Create(type, isNullable, CustomModifiers));
+                }
+            }
+            return (false, default);
```

Comment:
>default [](start = 27, length = 7)

I don't think there is a scenario that can fail (but it could warn). We should compute and return a computed merged type always.